src/libraries/System.Security.Cryptography.Xml/tests/SignedXmlTest.cs

```diff
@@ -1637,5 +1637,97 @@ private void ValidateSignedXmlDefaultHashAlgorithms(string expectedSignatureMeth
                     xp.SelectSingleNode("/ds:SignedInfo/ds:Reference/ds:DigestMethod/@Algorithm", nsMgr)?.Value);
             }
         }
+
+        [Theory]
+        [InlineData("a" /*, 1*/)]
+        [InlineData("b" /*, 2*/)]
+        [InlineData("y" /*, 1*/)]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "SignedXml has been failing validation on nested signatures all the time with .NET Framework and .NET (Core) up to .NET 6. This test was added together with a fix for .NET 7.")]
+        public void CheckSignatureMultipleEnvelopedSignatures(string signatureParent/*, int expectedPosition*/)
+        {
+            // To reduce running time, the test data is a pre-calculated string. For anyone that want to
+            // make adjustments to it, this is the small program that was used to generate the data.
+
+            //void Main()
+            //{
+            //	var xml = "<root><x ID=\"a\"/><x ID=\"b\"><x ID=\"c\"><x ID=\"y\"/></x></x></root>";
+            //
+            //	var xd = new XmlDocument();
+            //	xd.LoadXml(xml);
+            //
+            //	Sign(xd, "c", "y");
+            //	Sign(xd, "b", "b");
+            //	Sign(xd, "a", "a");
+            //}
+            //
+            //void Sign(XmlDocument xd, string id, string signaturePlacement)
+            //{
+            //	var sx = new SignedXml(xd);
+            //	var key = RSA.Create();
+            //	sx.SigningKey = key;
+            //
+            //	var reference = new Reference("#" + id);
+            //	reference.AddTransform(new XmlDsigEnvelopedSignatureTransform());
+            //	reference.AddTransform(new XmlDsigExcC14NTransform());
+            //	sx.AddReference(reference);
+            //
+            //	sx.ComputeSignature();
+            //	sx.KeyInfo.AddClause(new RSAKeyValue(key));
+            //
+            //	var node = xd.SelectSingleNode("//x[@ID=\'" + signaturePlacement + "']");
+            //
+            //	var signatureElement = sx.GetXml();
+            //
+            //	node.AppendChild(sx.GetXml());
+            //}
```

Comment:
Why not calculate it once and assign in a static field?

Reply:
I think I prefer the approach of using the pregenerated value.  Makes it easier to understand what's going on, and is more like a conformance test.

Reply:
I agree with @bartonjs (Because he agrees with me ;) I think that the pregenerated value as it is makes it easier to read the code and understand the test. Including the generation code as a comment is a convenience to any future programmer that want to alter the test case, so they can easily reproduce the current generated data.